Balasana: Add Color Annotations

kjellr opened the pull request:
Follows the mold of #1944 to add basic color annotations for Balasana. 

![Screen Shot 2020-04-29 at 14 05 46](https://user-images.githubusercontent.com/1202812/80630651-8ac66380-8a22-11ea-94eb-9c87e921d0ce.png)

Still needs: 

- [x] Refinement around contrast. 
- [x] Double checking mobile styles.
- [x] Add featured color palettes.

kjellr committed:
Add basic color annotations

kjellr committed:
Add basic color annotations

kjellr committed:
Update themecolors in wpcom.php

kjellr committed:
Update themecolors in wpcom.php

kjellr committed:
Remove contrast annotations.

kjellr committed:
Remove contrast annotations.

kjellr committed:
Add grays

kjellr committed:
Add grays

allancole commented on balasana/inc/wpcom-colors.php:
```diff
@@ -0,0 +1,340 @@
+  * Custom CSS
+  */
+function balasana_custom_colors_extra_css() { ?>
+	::selection {
+		background-color: currentColor;
+		mix-blend-mode: hard-light;
+	}
+	::-moz-selection {
+		background-color: currentColor;
+		mix-blend-mode: hard-light;
+	}
```
So these selection colors don’t always come out that great. Plus browser support for `mix-blend-mode` isn’t great either. Wondering if we actually _need_ selection colors at all. It’s a nice subtle design detail, but without access to color opacity, it’s tough to really get it right without doing it manually for each theme. 

Varia now uses the primary and background colors for selected text like this:
https://github.com/Automattic/themes/pull/1944/files#diff-aa2f4bd1a0647c2da27f37eca952a82fR375-R392

It looks okay but it’s also kind of aggressive depending on the selected color options: 
![image](https://user-images.githubusercontent.com/709581/81752567-ae0bfc80-947f-11ea-9d72-a7bf1cef7376.png)

kjellr replied:
Yeah, maybe we should just set all of the selection styles back to `initial`.

kjellr replied:
FWIW, `mix-blend-mode` doesn't seem to work at all in Firefox, making the text unreadable.

kjellr committed:
Add featured color palettes

kjellr committed:
Add featured color palettes

kjellr committed:
Remove custom selection colors.

kjellr committed:
Remove custom selection colors.

allancole committed:
Balasana: Recompile CSS to remove !important from utility classes, update npm dependency.

allancole committed:
Balasana: Recompile CSS to remove !important from utility classes, update npm dependency.

allancole committed:
Balasana: Connect editor color palette to color annotations.

allancole committed:
Balasana: Connect editor color palette to color annotations.

allancole committed:
Balasana: Revise conditionals to fix array offset issue, cody tidying.

allancole committed:
Balasana: Revise conditionals to fix array offset issue, cody tidying.

allancole committed:
Balasana: Add editor color palette.

allancole committed:
Balasana: Add editor color palette.

allancole committed:
Balasana: Add editor-specific color annotations.
- Replace `white` named color references with `#ffffff` for consitency.
- Code tidying.

allancole committed:
Balasana: Add editor-specific color annotations.
- Replace `white` named color references with `#ffffff` for consitency.
- Code tidying.

allancole committed:
Balasana: Remove `!important` flags from utility classes.

allancole committed:
Balasana: Remove `!important` flags from utility classes.

allancole committed:
Balasana: Fix txt/bg text color.

allancole committed:
Balasana: Fix txt/bg text color.

allancole committed:
Balasana: Fix rtl nested button text-color, code tidying.

allancole committed:
Balasana: Fix rtl nested button text-color, code tidying.

allancole committed:
Balasana: Fix nested button text-color, code tidying.

allancole committed:
Balasana: Fix nested button text-color, code tidying.

allancole committed:
Balasana: Recompile styles with the latest varia changes

- Add `.background-dim` support
- Fix Button color selector specificity
- Add table block support
- Clean up cover block styles

allancole committed:
Balasana: Recompile styles with the latest varia changes

- Add `.background-dim` support
- Fix Button color selector specificity
- Add table block support
- Clean up cover block styles

allancole committed:
Balasana: Recompile RTL styles with the latest varia changes

- Add `.background-dim` support
- Fix Button color selector specificity
- Add table block support
- Clean up cover block styles

allancole committed:
Balasana: Recompile RTL styles with the latest varia changes

- Add `.background-dim` support
- Fix Button color selector specificity
- Add table block support
- Clean up cover block styles

allancole committed:
Balasana: Recompile editor-styles with the latest varia changes

- Add `.background-dim` support
- Fix Button color selector specificity
- Add table block support
- Clean up cover block styles

allancole committed:
Balasana: Recompile editor-styles with the latest varia changes

- Add `.background-dim` support
- Fix Button color selector specificity
- Add table block support
- Clean up cover block styles

allancole committed:
Balasana: Recompile styles against master and update color name.

allancole committed:
Balasana: Recompile styles against master and update color name.

allancole committed:
Balasana: Add color options for foreground-dark/light.
- Fix hidden editor color selectors

allancole committed:
Balasana: Add color options for foreground-dark/light.
- Fix hidden editor color selectors

allancole committed:
Balasana: Code tidying.

allancole committed:
Balasana: Code tidying.

allancole committed:
Balasana: Improving color annotations.

allancole committed:
Balasana: Improving color annotations.

allancole committed:
Merge branch 'add/color-annotations-balasana' of https://github.com/Automattic/themes into add/color-annotations-balasana

allancole committed:
Merge branch 'master' into add/color-annotations-balasana

allancole committed:
Balasana: Clean up primary hover color styles.

allancole committed:
Merge branch 'master' into add/color-annotations-balasana-rebase

allancole committed:
Rebase and recompile styles.

allancole committed:
Merge branch 'master' into add/color-annotations-balasana

allancole committed:
Balasana: Add underline to links inside blocks with a background-color.

allancole committed:
Merge branch 'master' into add/color-annotations-balasana

allancole committed:
Balasana: Clean up button colors, and fix background-background-color defaults in ALves template.

allancole committed:
Balasana: Fixes foreground-light/dark background colors.

allancole committed:
Balasana: Fix button underlines.

allancole committed:
Merge branch 'master' into add/color-annotations-balasana

allancole committed:
Balasana: Fix inconsistent text and link colors inside of group blocks with background colors

- Kudos to @kjellr
